crates/ruff_linter/src/settings/types.rs

```diff
@@ -62,6 +62,13 @@ impl From<PythonVersion> for Pep440Version {
     }
 }
 
+impl From<PythonVersion> for ruff_python_syntax_errors::PythonVersion {
+    fn from(value: PythonVersion) -> Self {
+        let (major, minor) = value.as_tuple();
+        Self { major, minor }
+    }
+}
```

Comment:
Oh no, more `PythonVersion` types :( I'm somewhat inclined to move red knot's `PythonVersion` to `ruff_db` and reuse that instead (Ruff already depends on the `ruff_db` crate so adding that dependency won't hurt.